### PR TITLE
Add submission narrative models/endpoints

### PR DESCRIPTION
**Pull request by cmc333333**
Users would like to be able to add notes ("narratives") about their files. We'll allow appropriate users to do this. This uses a more RESTful set of endpoints, but I'm happy to remove if we think it's better to keep the existing, imperative style.

Easiest to review changeset by changeset

### CODE REVIEW

File: dataactbroker/README.md [hunk 1]
```diff
@@ -843,6 +843,58 @@ Get total obligations and specific obligations. Calls to this route should inclu
...
+{}
+```
+
+
+#### GET "/v1/submission/<int:submission_id>/narrative"
+Retrieve existing submission narratives (explanations/notes for particular
+files). Submission id should be the integer id associated with the submission
+in question. Users must have appropriate permissions to access these
+narratives (write access for the agency of the submission, or SYS).
+
```
Comment (alburde1) [inline dataactbroker/README.md]: Is this a duplicate of the section starting at line 847 without the response?

**Comment by alburde1** [comment]
I see no problem with changing the style of API calls but I also haven't gotten super used to the old style so it's not a hard change for me. Might be good to get a second opinion.

File: dataactbroker/fileRoutes.py [hunk 2]
```diff
@@ -138,3 +139,17 @@ def sign_submission_file():
         fileManager = FileHandler(request, isLocal=IS_LOCAL, serverPath=SERVER_PATH)
         fileManager.addInterfaces(InterfaceHolder())    # soon to be removed
         return fileManager.get_signed_url_for_submission_file()
+
+    @app.route("/v1/submission/<int:submission_id>/narrative", methods=['GET'])
```
Comment (bsweger) [inline dataactbroker/fileRoutes.py]: Thanks for nudging us in this direction!

File: dataactbroker/handlers/fileHandler.py [hunk 3]
```diff
@@ -41,6 +43,30 @@
 _smx_logger = logging.getLogger('deprecated.smx')
 
 
+def user_agency_matches(submission):
```
Comment (bsweger) [inline dataactbroker/handlers/fileHandler.py]: much better as a separate function--thanks!

File: tests/unit/dataactbroker/test_file_handler.py [hunk 4]
```diff
@@ -120,3 +124,87 @@ def test_list_submissions_failure(database, job_constants, monkeypatch):
...
+    fn(sub.submission_id)
+
+    monkeypatch.setattr(fileHandler, 'user_agency_matches',
+                        Mock(return_value=False))
+    with pytest.raises(ResponseException):
+        fn(sub.submission_id)
+
+
+def test_narratives(database, job_constants, monkeypatch):
+    """Integration test that verifies we can add, retrieve, and update
```
Comment (bsweger) [inline tests/unit/dataactbroker/test_file_handler.py]: Is it important to retain our current distinction, location-wise, between unit and integration tests?

Comment (cmc333333) [inline tests/unit/dataactbroker/test_file_handler.py]: Eh, I don't have clear definitions, so I'm probably using the term incorrectly. I meant "this isn't testing a single unit of work, it's testing a few at once". I can swap out to be more explicit if that'd be better?

**Comment by bsweger** [comment]
@cmc333333 This looks good to me--I'm ready to merge if you are. Not for this PR, but should we consider telling CodeClimate to ignore the Alembic migrations?

**Comment by cmc333333** [comment]
Thanks @bsweger! Updated a docstring. I agree about code climate; I started work on a config but saw all the potential options and punted :)

**Comment by bsweger** [comment]
Thanks for the doctring clarification! Merging

**Pull request merged**